core/server/common/src/main/java/alluxio/master/AbstractMaster.java

```diff
@@ -133,12 +132,9 @@ protected ExecutorService getExecutorService() {
 
   @Override
   public JournalContext createJournalContext() throws UnavailableException {
-    // All modifications to journaled state must happen inside of a journal context so that we can
-    // persist the state change. As a mechanism to allow for state pauses, we acquire the state
-    // change lock before entering any code paths that could modify journaled state.
-    try (LockResource l = new LockResource(mMasterContext.stateChangeLock())) {
-      return mJournal.createJournalContext();
-    }
+    // Use the state change lock for the journal context, since all modifications to journaled
+    // state must happen inside of a journal context.
+    return mJournal.createJournalContext(mMasterContext.stateChangeLock());
```

Comment:
Can't we leave the Journal interfaces the same but wrap context here with a forwarder context that holds the lock. 
So Journal implementations wouldn't need to manage the lock.

Reply:
I added a new journal context for the state change lock